addons/website_sale/models/website_snippet_filter.py

```diff
@@ -1,39 +1,140 @@
+                'image_512': b'/product/static/img/product_lamp.png',
                 'display_name': _('Lamp'),
                 'description_sale': _('Lightbulb sold separately'),
             }, {
-                'image_512': '/product/static/img/product_product_20-image.png',
+                'image_512': b'/product/static/img/product_product_20-image.png',
                 'display_name': _('Whiteboard'),
                 'description_sale': _('With three feet'),
             }, {
-                'image_512': '/product/static/img/product_product_27-image.png',
+                'image_512': b'/product/static/img/product_product_27-image.png',
                 'display_name': _('Drawer'),
                 'description_sale': _('On wheels'),
+            }, {
+                'image_512': b'/product/static/img/product_product_7-image.png',
+                'display_name': _('Box'),
+                'description_sale': _('Reinforced for heavy loads'),
+            }, {
+                'image_512': b'/product/static/img/product_product_9-image.png',
+                'display_name': _('Bin'),
+                'description_sale': _('Pedal-based opening system'),
             }]
             merged = []
             for index in range(0, max(len(samples), len(data))):
                 merged.append({**samples[index % len(samples)], **data[index % len(data)]})
                 # merge definitions
             samples = merged
         return samples
+
+    def _filter_records_to_values(self, records, is_sample=False):
+        res_products = super()._filter_records_to_values(records, is_sample)
+        if self._get_model()._name == 'product.product':
+            max_nb_chars = 100
+            for res_product in res_products:
+                product = res_product.get('_record')
+                if not is_sample:
+                    res_product.update(product._get_combination_info_variant())
+                if res_product['description_sale'] and len(res_product['description_sale']) > max_nb_chars:
+                    res_product['description_sale'] = "%s ..." % res_product['description_sale'][:max_nb_chars]
+        return res_products
+
+    @api.model
+    def _get_products(self, mode, context):
+        dynamic_filter = context.get('dynamic_filter')
+        handler = getattr(self, '_get_products_%s' % mode, self._get_products_latest_sold)
+        website = self.env['website'].get_current_website()
+        search_domain = context.get('search_domain')
+        limit = context.get('limit')
+        domain = [('website_published', '=', True)] + website.website_domain() + (search_domain or [])
+        products = handler(website, limit, domain, context)
+        return dynamic_filter._filter_records_to_values(products, False)
+
+    def _get_products_latest_sold(self, website, limit, domain, context):
+        products = []
+        sale_orders = self.env['sale.order'].sudo().search(website.website_domain() + [
```

Comment:
website_domain -> website_id = X

We don't want backend sales 

to apply in others method too